### PR TITLE
11 add food serversideroute db function

### LINKED ISSUE #11: Create server-side route + db function to add a new food
Create database function (`addFood(food)`) and server-side POST route (`/api/v1/foods/:id`) to add details for a new food, as [described in the Wiki](https://github.com/matai-2019/foodlist/wiki/Server-Routes).

### PR DESCRIPTION
**Pull request by Rahul-Lal**
Closes #11

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (emilyparkes) [inline .vscode/settings.json]: Exclude this file from changes pushed to development please :)

File: server/db/foods.js [hunk 2]
```diff
@@ -4,6 +4,22 @@ function getFoods (db = connection) {
   return db('foods')
 }
 
+function addFood (food, db = connection) {
+  let foodID
+
```
Comment (emilyparkes) [inline server/db/foods.js]: No new line here please :)

Comment (emilyparkes) [reply server/db/foods.js]: Also could we actually move this function under the `getFood` function?

File: server/db/foods.js [hunk 3]
```diff
@@ -22,5 +38,6 @@ function getFood (id, db = connection) {
 
 module.exports = {
   getFoods,
+  addFood,
```
Comment (emilyparkes) [inline server/db/foods.js:47]: Swap order of this with the below fn. Thanks!

File: server/routes/foods.js [hunk 4]
```diff
@@ -9,6 +9,15 @@ router.get('/', (req, res) => {
     .then(foods => res.status(200).json(foods))
 })
 
+router.post('/', (req, res) => {
```
Comment (emilyparkes) [inline server/routes/foods.js:23]: Same here all edit routes below the get routes please :)

**Review by emilyparkes — CHANGES_REQUESTED** [review]
Could you move all your post routes and db functions to be below the get routes and function in the same file please?

File: .vscode/settings.json (hunk 1 above)
Comment (Rahul-Lal) [reply .vscode/settings.json]: Cool, get on it now. Gracias!

File: server/routes/foods.js (hunk 4 above)
Comment (Rahul-Lal) [reply server/routes/foods.js:23]: done

File: server/db/foods.js (hunk 3 above)
Comment (Rahul-Lal) [reply server/db/foods.js:47]: done

File: server/db/foods.js (hunk 2 above)
Comment (Rahul-Lal) [reply server/db/foods.js]: done

File: .vscode/settings.json (hunk 1 above)
Comment (Rahul-Lal) [reply .vscode/settings.json]: done

File: tests/server/db/foods.test.js [hunk 5]
```diff
@@ -24,6 +24,21 @@ test('db.getFoods returns an array of 27 foods', () => {
   //   })
 })
 
+test('db.addFood should add food to db', () => {
```
Comment (emilyparkes) [inline tests/server/db/foods.test.js:47]: Can this be underneath `getFood` too please?
Just for consistency and so our simpler fns are at the top. Cheers 👍

File: tests/server/routes/foods.test.js [hunk 6]
```diff
@@ -1,11 +1,13 @@
 import request from 'supertest'
+import { Item } from 'semantic-ui-react'
 
 jest.mock('../../../server/db/foods', () => ({
   getFoods: () => Promise.resolve([
     { id: 1, name: 'Lamb' },
     { id: 2, name: 'Beef' },
     { id: 3, name: 'Broccoli' }
   ]),
+  addFood: () => Promise.resolve([4]),
```
Comment (emilyparkes) [inline tests/server/routes/foods.test.js]: Move this down please :)

File: tests/server/routes/foods.test.js [hunk 7]
```diff
@@ -48,6 +51,22 @@ test('GET /foods returns all of the foods', () => {
     })
 })
 
+test('POST adds a new food', () => {
```
Comment (emilyparkes) [inline tests/server/routes/foods.test.js:67]: Move this down please :)

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: tests/server/routes/foods.test.js (hunk 6 above)
Comment (Rahul-Lal) [reply tests/server/routes/foods.test.js]: Which line?

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**